darwin/exporter/formats/nifti.py

```diff
@@ -71,87 +132,99 @@ def export_single_nifti_file(video_annotation: dt.AnnotationFile, output_dir: Pa
-    # Builds output volumes per class
-    if output_volumes is None:
-        output_volumes = {class_name: np.zeros(volume_dims) for class_name in class_map.keys()}
-    # Loops through annotations to build volumes
-    for _, annotation in enumerate(video_annotation.annotations):
-        frames = annotation.frames
-        for frame_idx in frames.keys():
-            view_idx = get_view_idx_from_slot_name(annotation.slot_names[0])
-            if view_idx == 0:
-                height, width = volume_dims[0], volume_dims[1]
-                pixdims = [pixdim[0], pixdim[1]]
-            elif view_idx == 1:
-                height, width = volume_dims[0], volume_dims[2]
-                pixdims = [pixdim[0], pixdim[2]]
-            elif view_idx == 2:
-                height, width = volume_dims[1], volume_dims[2]
-                pixdims = [pixdim[1], pixdim[2]]
-            if "paths" in frames[frame_idx].data:
-                # Dealing with a complex polygon
-                polygons = [shift_polygon_coords(polygon_path) for polygon_path in frames[frame_idx].data["paths"]]
-            elif "path" in frames[frame_idx].data:
-                # Dealing with a simple polygon
-                polygons = shift_polygon_coords(frames[frame_idx].data["path"])
+        return create_error_message_json("video_annotation not found", output_dir, image_id)  
+    
+    for slot in video_annotation.slots:
+        # Pick the first slot to take the metadata from. We assume that all slots have the same metadata.
+        metadata = slot.metadata
+        if metadata is None:
+            return create_error_message_json(
+                f"No metadata found for {str(filename)}, are you sure this is medical data?", output_dir, image_id
+            )
+
+        volume_dims, pixdim, affine, _ = process_metadata(metadata)
+        if affine is None or pixdim is None or volume_dims is None:
+            return create_error_message_json(
+                f"Missing one of affine, pixdim or shape in metadata for {str(filename)}, try reuploading file",
+                output_dir,
+                image_id,
+            )
+    return image_id
+
+
+def populate_output_volumes(annotation: Union[dt.Annotation, dt.VideoAnnotation], output_dir: str, slot_map: Dict, output_volumes: Dict, image_id: str) -> None:
+
+    slot_name = annotation.slot_names[0]
+    slot = slot_map[slot_name]
+    series_instance_uid = slot.metadata.get("SeriesInstanceUID", "SeriesIntanceUIDNotProvided")
+    volume = output_volumes.get(series_instance_uid)
+    volume = output_volumes.get(series_instance_uid)
+    frames = annotation.frames
+    frame_new = {}        
+    for frame_idx in frames.keys():
+        frame_new[frame_idx] = frames
+        view_idx = get_view_idx_from_slot_name(slot_name, slot.metadata.get("orientation"))
+        if view_idx == 0:
+            height, width = volume[annotation.annotation_class.name].dims[0], volume[annotation.annotation_class.name].dims[1]
```

Comment:
I don't often suggest comments, but in this case, I think that the clarification of looking at the X-Y, X-Z, or Y-Z planes might be warranted for maintainers of the code.  It's much better already though

Reply:
Also, if you wanted to avoid comments, and use a different solution, feel free.  Could for instance do
```
XYPLANE = 0
```
and then

```
if view_idx == XYPLANE
```

might also want to include the medical terms (maybe saggital, coronal, axial), although given that there are several ways to describe these, and several affines from which a radiological plane can be observed, it gets...messy!